### PR TITLE
[DataMovement] Removed double JobPartEvent being added and updated all tests to track events raised

### PR DESCRIPTION
**Pull request by amnguye**
This change removes the JobPartEvent being added on listing jobs in the job parts since it's already done on the JobPart constructor level.

This change also adds the check of the expected events to be raised on all tests.

Not to be merged for now as there's test flakiness. Due to the SingleTransferEvents being invoked after the completion status of the overall job has been raised.

### CODE REVIEW

File: sdk/storage/Azure.Storage.DataMovement/tests/TestEventsRaised.cs [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+        /// or failures.
+        /// </summary>
+        /// <param name="blobCount">
+        /// Expected amount of single transfers within the container transfer.
+        /// </param>
+        public void AssertContainerCompletedCheck(int transferCount)
+        {
+            AssertUnexpectedFailureCheck();
+            Assert.IsEmpty(SkippedEvents);
+            Assert.AreEqual(transferCount, SingleCompletedEvents.Count);
```
Comment (amnguye) [inline sdk/storage/Azure.Storage.DataMovement/tests/TestEventsRaised.cs:180]: This line is causing the flakiness to occur. Going to disable this for now until we can circle back on a solution on the flakiness of a completed state and getting back all our SingleCompletedEvents.

Comment (amnguye) [reply sdk/storage/Azure.Storage.DataMovement/tests/TestEventsRaised.cs:180]: https://github.com/Azure/azure-sdk-for-net/issues/35976

**Review by jalauzon-msft — APPROVED** [review]
Should probably run live tests just to make sure removing this JobPartEvent did not break anything :)

**Comment by amnguye** [comment]
The live tests failure for mac os are unrelated to this PR

**Pull request merged**